tensorflow/compiler/tf2xla/kernels/bincount_op.cc

```diff
@@ -0,0 +1,98 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/compiler/xla/shape_util.h"
+#include "tensorflow/compiler/tf2xla/xla_op_kernel.h"
+#include "tensorflow/compiler/tf2xla/xla_op_registry.h"
+#include "tensorflow/compiler/tf2xla/xla_helpers.h"
+#include "tensorflow/compiler/xla/client/lib/comparators.h"
+#include "tensorflow/compiler/xla/client/lib/constants.h"
+#include "tensorflow/compiler/xla/client/lib/arithmetic.h"
+#include "tensorflow/compiler/xla/client/xla_computation.h"
+
+namespace tensorflow {
+namespace {
+
+// TODO: This is only a dummy kernel
+class BincountOp : public XlaOpKernel {
+ public:
+  explicit BincountOp(OpKernelConstruction* ctx) : XlaOpKernel(ctx) {
+    DataType dtype;
+  }
+  
+  void Compile(XlaOpKernelContext* ctx) override {
+  // Dumb implementation for the simplest test case
+    xla::XlaOp input = ctx->Input(0);
+    auto max = xla::ReduceAll(
+        input, xla::Zero(ctx->builder(), xla::S32),
+        xla::CreateScalarMaxComputation(xla::S32, ctx->builder()));
+    // TODO: it need to be max
+    int64_t output_size = 6;
```

Comment:
As max is an `xlaOp` how we could retrieve the value to use it in the output_size.

Reply:
It needs to be bound by largest possible output size

Reply:
The problem is that without params the output size is `tf.reduce_max` on the input (it is prepared in the python API of the op at runtime) so It rely on the input content.
Do you want to use `std::numeric_limits` here and then usa a negative pad at the output?

Reply:
Ok I've used register as `CompileTimeConstantInput` for the `size` param.